src/__tests__/generate.test.ts

```diff
@@ -0,0 +1,149 @@
+    } catch (error) {
+      console.error(error);
+    }
+  });
+
+  afterAll(async () => {
+    // delete our temporary directories
+    try {
+      await removeTempDir(tempRoot);
+      await removeTempDir(`${root}/_templates`);
+    } catch (err) {
+      console.error(err);
+    }
+  });
+
+  it("expects to find a temporary directory in src/components", async () => {
+    const tempDirectoryExists = await checkDirExists(tempRoot);
+
+    expect(tempDirectoryExists).toBe(true);
+  });
+
+  it("works without flags", async () => {
+    const componentName = DEFAULT_COMPONENT_NAME;
+    const componentFolderPath = `${tempRoot}/${componentName}`;
+
+    execSync(`./bin/run generate`, {
+      cwd: root
+    });
+
+    const newComponentFolderExists = await checkDirExists(componentFolderPath);
+    const componentIndexExists = checkFileExists(
+      `${componentFolderPath}/index.js`
+    );
+
+    expect(newComponentFolderExists).toBe(true);
+    expect(componentIndexExists).toBe(true);
+  });
+
+  it("works with a flag of a valid template flag", async () => {
+    const componentName = "TestComponent";
+    const componentFolderPath = `${tempRoot}/${componentName}`;
+
+    execSync(`./bin/run generate -t react-component -n ${componentName}`, {
+      cwd: root
+    });
+
+    const newComponentFolderExists = await checkDirExists(componentFolderPath);
+    const componentIndexExists = checkFileExists(
+      `${componentFolderPath}/index.js`
+    );
+
+    expect(newComponentFolderExists).toBe(true);
+    expect(componentIndexExists).toBe(true);
+  });
+
+  it("uses the default templates if none in the users directory", async () => {
+    const componentName = "DefaultTemplateComponent";
+    const componentFolderPath = `${tempRoot}/${componentName}`;
+
+    execSync(`./bin/run generate -t react-component -n ${componentName}`, {
+      cwd: root
+    });
+
+    const newComponentFolderExists = await checkDirExists(componentFolderPath);
+    const componentIndexExists = checkFileExists(
+      `${componentFolderPath}/index.js`
+    );
+
+    expect(newComponentFolderExists).toBe(true);
+    expect(componentIndexExists).toBe(true);
+  });
+
+  it("throws an error when you pass an invalid flag", () => {
+    const generateCommand = () =>
+      execSync(`./bin/run generate -t fake-component -n FakeComponent`, {
+        cwd: root
+      });
+
+    expect(generateCommand).toThrowErrorMatchingSnapshot();
+  });
+
+  it("uses the users template if they have one", async () => {
+    const pathToNewTemplate = `${root}/_templates/react-component/new`;
+    const newComponentName = "Test";
+    const newTemplate = `---\nto: <%= path %>/src/components/<%= h.changeCase.pascal(name) %>.jsx
+    ---
+    <% component = h.changeCase.pascal(name) -%>
+    import React from 'react'
+
+    /** Description of component */
+    export const <%= component %> = () => {
+      return (
+      <h1>Hello <%= component %> component!</h1>
+      )
+    }`;
```

Comment:
Should we create a templats directory in a `tests/helpers` directory to keep the test "clean"?

Reply:
Oh wait, this looks outdated - I moved this to a separate file in utils, but let me instead move it to `tests/helpers` like you suggest. Good idea!

Reply:
Done.